lib/rubocop/cop/ezcater/migration/bigint_foreign_key.rb

```diff
@@ -0,0 +1,264 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Ezcater
+      module Migration
+        # Use `bigint` instead of `integer` for all foreign keys.
+        #
+        # @example
+        #
+        #   # bad
+        #   create_table :foos do |t|
+        #     t.integer :bar_id
+        #   end
+        #
+        #   # bad
+        #   create_table :foos do |t|
+        #     t.integer :bar_id, limit: 7
+        #   end
+        #
+        #   # bad
+        #   add_column :foos, :bar_id, :integer
+        #
+        #   # bad
+        #   add_column :foos, :bar_id, :integer, limit: 7
+        #
+        #   # bad
+        #   add_reference :foos, :bar, type: :integer
+        #
+        #   # bad
+        #   add_reference :foos, :bar, type: :integer, limit: 7
+        #
+        #   # good
+        #   create_table :foos do |t|
+        #     t.bigint :bar_id
+        #   end
+        #
+        #   # good
+        #   create_table :foos do |t|
+        #     t.references :bar
+        #   end
+        #
+        #   # good
+        #   add_column :foos, :bar_id, :bigint
+        #
+        #   # good
+        #   add_reference :foos, :bar
+        #
+        class BigintForeignKey < Base # rubocop:disable Metrics/ClassLength
+          extend AutoCorrector
+
+          MSG = <<~MSG.chomp
+            Use `bigint` instead of `integer` for foreign keys. This prevents them from potentially running out of int values before their referenced primary keys.
```

Comment:
Thoughts on this slight language change for clarity?:

```suggestion
            Use `bigint` instead of `integer` for foreign keys. This ensures that they are capable of storing values from referenced primary keys which are `bigint` or may eventually be migrated to `bigint`.
```